src/QsCompiler/CompilationManager/EditorSupport/EditorCompletion.cs

```diff
@@ -661,5 +676,17 @@ private static Position GetDelimiterPosition(FileContentManager file, CodeFragme
                 IsIncomplete = isIncomplete,
                 Items = items?.ToArray()
             };
+
+        /// <summary>
+        /// Returns true if the qualified name is visible given the current namespace and a list of open namespaces.
+        /// <para/>
+        /// Names that start with "_" are treated as "private"; they are only visible from the namespace in which they
+        /// are declared.
+        /// </summary>
+        private static bool IsVisible(QsQualifiedName qualifiedName,
+                                      string currentNamespace,
+                                      IEnumerable<string> openNamespaces) =>
+            openNamespaces.Contains(qualifiedName.Namespace.Value) &&
+            (!qualifiedName.Name.Value.StartsWith("_") || qualifiedName.Namespace.Value == currentNamespace);
```

Comment:
To sum up things: any types or callables starting with an underscore are suggested for completions "anywhere" (depending on context) within the same namespace as they are declared in, possibly across compilation units, correct? And not in any other namespace, independent on file or compilation unit. I think once we actually properly handle that with access modifiers in the language we will probably restrict it to visible within the compilation unit, but at that point the completion here will automatically pick it up. So, looks good to me. :)

Reply:
Yes, that's correct, it's only based on having the same parent namespace regardless of file or compilation unit.